spec/requests/host_aggregates_spec.rb

```diff
@@ -67,6 +68,61 @@
     end
   end
 
+  describe 'POST /api/host_aggregates/:id' do
+
+    before do
+      @zone = FactoryBot.create(:zone, :name => "api_zone")
+      @microstack = FactoryBot.create(:ems_cloud)
+      @provider = FactoryBot.create(:provider, :zone => @zone)
+      @microstack.update(:provider => @provider) ## microstack.provider = provider 
+      @infra = FactoryBot.create(:ems_infra, :name => "microstack ironic", :hostname => "ironic", :zone => @zone, :provider => @provider)
+      @cluster = FactoryBot.create(:ems_cluster, :name => "Microstack Compute", :ext_management_system => @infra)
+      @host = FactoryBot.create(:host_with_authentication, :ext_management_system => @infra, :ems_cluster => @cluster, :name => "host", :hostname => "host.ironic.microstack", :vmm_vendor => "openstack_infra")
+    end
```

Comment:
All of this setup was needed to test the UI but strictly for the API specs you don't need all of this.

Was able to get this working with just:
```suggestion
    let(:zone) { FactoryBot.create(:zone) }
    let(:ems)  { FactoryBot.create(:ems_cloud, :zone => zone) }
    let(:host) { FactoryBot.create(:host, :ext_management_system => ems) }
```